docs/src/pages/guides/api/api.md

```diff
@@ -136,3 +136,10 @@ The Material-UI components use a combination of the two approaches according to
 - An *enum* is used when **> 2** degrees of freedom are required, or if there is the possibility that additional degrees of freedom may be required in the future.
 
 Going back to the previous button example; since it requires 3 degrees of freedom, we use an *enum*.
+
+## Glossary
+
+- **BFS**: Breadth-first search.
+- **host element**: a DOM node in the context of `react-dom`.
```

Comment:
```suggestion
- **host components**: represents a DOM node. See also [React Implementation Notes](https://reactjs.org/docs/implementation-notes.html#mounting-host-elements).
```

The context results in it being a DOM node. It would be a string for the server renderer. Listing the result and the renderer is a bit redundant.

element and component would both work. There's maybe some subtle distinction between "pass props to a component" vs. pass props to an element. I think people are more familiar with the component term I suppose.

Reply:
https://overreacted.io/react-as-a-ui-runtime/ Maybe we should use the "host instance" wording?

Reply:
Isn't the instance what's created during render i.e. the Component holds the implementation and the instance is well an instance of that component. `div` is a host component but what you return from render is a host element `<div />` and the host element has a 1:1 mapping to a host instance.  

I think we you can explain that without the notion of "host":
- `Component` <- the **component** (may extend React.Component or a function or a string
- `<Component />` <- the **element** i.e. the return value from `React.createElement(Component)`. These can be created at any time but usually in a render function
- `<Component ref={ref} />` <- ref.current will point to the component **instance**. For class components it's an instance of the class, for function components it's `null`, for host components it's an instance of `Element`

Ergo props spreading to an instance doesn't make sense at all. Spreading it to an element would mean you spread it into the return value from `createElement` but that's not what you want to do. You want it in the call to `createElement` i.e. `createElement(Component, { ...other });`

Reply:
> Isn't the instance what's created during render 

Why when the object is created important?

We should change the documentation wording then.

Reply:
> > Isn't the instance what's created during render
> 
> Why when the object is created important?
> 
> We should change the API documentation wording then.

Forget the first part. The latter distinction should be more understandable.